### ISSUE TITLE
In POSIX sh, type flags are undefined

### ISSUE THREAD

**Issue by contivero** [koalaman/shellcheck#1471]
#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
type -a ls
```

#### Here's what shellcheck currently says:
Nothing

#### Here's what I wanted or expected to see:
"In POSIX sh, type flags are undefined"

There are no flags for the `type` builtin defined under POSIX sh ([see spec](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/type.html)), and the same is true for dash. Bash, however, allows [-aftpP].
